2_pvc_destination_gen/tasks/ensure_pvcs.yml

```diff
@@ -0,0 +1,46 @@
+- name: "Parsing pvc data"
+  set_fact:
+    mig_dest_pvc_data: []
+    mig_dest_adjusted_pvc_data: []
+    adjustment_report: []
+
+- name: "Convert storage class"
+  vars:
+    mig_converted_sc:
+      storage_class: "{{ mig_storage_class_mappings.get(item.storage_class+'_'+('RWX' if 'ReadWriteMany' in item.access_modes else ('ROX' if 'ReadOnlyMany' in item.access_modes else 'RWO')), item.storage_class)}}"
+  set_fact:
+    mig_dest_pvc_data: "{{ mig_dest_pvc_data + [(item | combine(mig_converted_sc, recursive=true))] }}"
+  loop: "{{ pvcs }}"
+
+- block:
+  - name: "Check if adjustment report file exists"
+    stat:
+      path: "{{ pv_size_adjustment_report_path }}"
+    register: adjustment_report_file
+
+  - name: "Reading volume size adjustment report"
+    set_fact:
+      adjustment_report: "{{ lookup('file', pv_size_adjustment_report_path) | from_json }}"
+    when: adjustment_report_file.stat.exists
+
+  - name: "Adjusting resulting sizes for applicable PVCs"
+    vars:
+      name_query: "[?pvc_name=='{{ original_pvc_data.pvc_name }}']"
```

Comment:
What's going on here? I haven't seen this type of expression

Reply:
@djwhatle this piece reads the PV size adjustment report generated by `stage_a` and finds out whether the current PVC is present in that report. If the PVC is found in the report, it uses the new adjusted size on the destination cluster.